src/ray/common/ray_config.h

```diff
@@ -23,27 +23,23 @@
 #include "ray/util/logging.h"
 
 class RayConfig {
-  // Needed for the RAY_CONFIG macro below to work properly for strings.
-  typedef std::string string_type;
-#define TOSTRING0(x) #x
-#define TOSTRING(x) TOSTRING0(x)
-
 /// -----------Include ray_config_def.h to define config items.----------------
 /// A helper macro that defines a config item.
 /// In particular, this generates a private field called `name_` and a public getter
 /// method called `name()` for a given config item.
 ///
 /// Configs defined in this way can be overriden by setting the env variable
-/// RAY_{name}=value where {name} is the variable name.
+/// RAY_{name}=value where {name} is the capitalized variable name.
 ///
 /// \param type Type of the config item.
 /// \param name Name of the config item.
 /// \param default_value Default value of the config item.
-#define RAY_CONFIG(type, name, default_value)                      \
- private:                                                          \
-  type name##_ = env_##type("RAY_" TOSTRING(name), default_value); \
-                                                                   \
- public:                                                           \
+#define RAY_CONFIG(type, name, default_value)                                     \
+ private:                                                                         \
+  type name##_ =                                                                  \
+      ReadEnv<type>("RAY_" + absl::AsciiStrToUpper(#name), #type, default_value); \
```

Comment:
Is it possible to keep the lower-case names? I think it emphasizes:
1. It's an internal system config
2. It's consistent with how gflags env vars work: https://gflags.github.io/gflags/

I'd like to reserve ALL_UPPER_CASE env vars for officially supported configs.